src/NumField/CM.jl

```diff
@@ -60,6 +60,10 @@ end
 Base.:(==)(f::CMType, g::CMType) = (f.field === g.field) &&
   issetequal(f.embeddings, g.embeddings)
 
+# this is not optimal, we would need to sort the embeddings on creation using
+# _absolute_index, but do we want this?
+Base.hash(f::CMType, h::UInt) = h
```

Comment:
```suggestion
Base.hash(f::CMType, h::UInt) = hash(f.field, h)
```
You can include the field here safely. Regarding the embeddings, you could just add `Set(f.embeddings)` to the hash, as that hashes the elements without any order (so the perfect analog to `issetequal`)

Reply:
Have you looked at the hashing of `Set`? I looks pretty wrong to me.

Reply:
Why? it hashes all elements on their own and then xors them. And xor is commutative, so order shouldn't matter

Reply:
ah, I missed the commutativity